### ISSUE TITLE
Update README

### ISSUE THREAD

**Issue by TGM** [Harold-Glitch/bitreverse#1]
The program is interesting. But you might want to add more details about usage and how it works.

What exactly are you generating and based on what? As far as I know eth and btc private keys are far from identical to be matches against btc and eth addresses in the same time.

**Comment by Harold-Glitch** [comment]
Yes I'll do. For the second part of the question Bitcoin (also Altcoins like Dash or LTC) and Ethereum have the secp256k1 part in common, this takes 50% of the GPU and at the end of computation we have to hash with compressed & uncompressed BTC algorithm and Keccak (Ethereum). Okay it is longer but we add 2'300'000 more addresses to target.

What's best ? 
- multiply the speed by 1.2 
- or add 2'300'000 chances to win

**Comment by TGM** [comment]
ETH - https://etherworld.co/2017/11/17/understanding-the-concept-of-private-key-public-key-and-address-in-ethereum-blockchain/
BTC - https://royalforkblog.github.io/2014/08/11/graphical-address-generator/#hello

The end-result seems pretty different.

I have privately implemented a similar project in nodejs, a few months ago. The common point between the currencies seems to elude me.

**Comment by Harold-Glitch** [comment]
Okay yes we deal with 3 different output hashes (addresses) the end result is different.
- secp256k1 compressed pubkey -> sha256 -> ripemd160 = BTC compressed address 
- secp256k1 uncompressed pubkey -> sha256 -> ripemd160 = BTC uncompressed address
- secp256k1 uncompressed pubkey -> keccac256 -> Ethereum address

The common point is secp256k1. Computation of secp256k1 is slow (even super optimized) comparing to the keccac256 hash, so the cost of adding Ethereum is not so high.

**Comment by TGM** [comment]
The performance is kind of slow tough, 44'000'000 -> 44KiloKeys/sec if I'm not mistaken.
I suppose you are loading the database in RAM. Are you using a one line array for match?

**Comment by Harold-Glitch** [comment]
GTX 1060 3GB it is exactly 48MegaKeys/secs, there is four nested level of bloom for matching with 18 Mega keys. GTX 1070 Ti runs at 71MegaKeys/secs. I have only valuable addresses > than 0.01 BTC or 0.1 ETH.

**Comment by TGM** [comment]
I like your style. :+1: I'll give it a go.
Try adding incremental ranges. We can setup a pool.

**Comment by TGM** [comment]
BTW - Have your tried to hookup in OpenGL? Should open compatibility with AMD cards.

**Comment by Harold-Glitch** [comment]
Hi!

Yes I know I coming from deep learning, I'm on the nvidia side. Got no AMD cards for testing right now. It should be a good option it I have a good playing field of AMD GPUs.

**Comment by TGM** [comment]
Once more thing. Can you add a switch for CPU or GPU?
--cpu (cpu only computing)
--gpu (gpu only computing)
Alternatively --gpu id

**Comment by Harold-Glitch** [comment]
--cpu is 20 times slower

**Comment by TGM** [comment]
What data is sent to bitreverse.io?

**Comment by Harold-Glitch** [comment]
I've my own data extraction rig. It is not complicated, addresses > 0.001 BTC (or for altcoin > 1) or EHT > 0.01. This is just addresses with relevant amount.

**Comment by TGM** [comment]
Can you add incremental variables as input?

**Comment by prodnet** [comment]
Can be installed and executed locally using your own address files without a remote connection?

**Comment by prodnet** [comment]
Can you repy to my question, please?
Thank you very much.

**Comment by Simkas45** [comment]
help me please run, not started programm windows 10

**Comment by xBreadyx** [comment]
Good luck guys, im worked😊

**Comment by MAITHI19** [comment]
how to install this program please help me
